logstash-core/lib/logstash/elasticsearch_client.rb

```diff
@@ -39,8 +39,11 @@ def initialize(settings, logger)
       @client_args[:ssl] = ssl_options
 
       username = @settings["var.elasticsearch.username"]
-      password = @settings["var.elasticsearch.password"]
       if username
+        password = @settings["var.elasticsearch.password"]
+        if password.is_a?(LogStash::Util::Password)
```

Comment:
What we should actually do is this 

`LogStash::Utils::Password(password)` and this would always return a Password object, this is similar to other rubyism.


```
Array([1, 2, 3]) => [1,2,3]
Array(1) => [1]
```

I _don't think_ we can get a NIL password an empty string yes but NIL no.